src/main/java/org/rumbledb/expressions/control/TypeSwitchCaseExpression.java

```diff
@@ -5,20 +5,20 @@
 import org.rumbledb.expressions.Expression;
 import org.rumbledb.expressions.Node;
 import org.rumbledb.expressions.flowr.FlworVarSequenceType;
-import org.rumbledb.expressions.primary.VariableReference;
+import org.rumbledb.expressions.primary.VariableReferenceExpression;
 import sparksoniq.semantics.visitor.AbstractNodeVisitor;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class TypeSwitchCaseExpression extends Expression {
 
-    private VariableReference variableReferenceNode;
+    private VariableReferenceExpression variableReferenceNode;
```

Comment:
Cosmetic suggestion: shall we remove 'node' keyword from this variable's name? (adding expression or not is optional)